src/shared/tpm2-util.c

```diff
@@ -630,9 +646,30 @@ static int tpm2_make_encryption_session(
                         TPMA_SESSION_CONTINUESESSION;
         ESYS_TR session = ESYS_TR_NONE;
         TSS2_RC rc;
-
+        TPM2B_AUTH auth = { 0 };
         assert(c);
 
+        /*
+         * if a pin is set for the seal object, use it to bind the session
+         * key to that object. This prevents active bus interposers from
+         * faking a TPM and seeing the unsealed value. An active interposer
+         * could fake a TPM, satisfying the encrypted session, and just
+         * forward everything to the *real* TPM.
+         */
+        if (pin) {
+            hash_pin(pin, strlen(pin), &auth);
+
+            rc = sym_Esys_TR_SetAuth(c, bind_key, &auth);
+            /* ESAPI knows about it, so clear it from our memory */
+            explicit_bzero_safe(&auth, sizeof(auth));
+            if (rc != TSS2_RC_SUCCESS) {
+                return log_error_errno(
+                    SYNTHETIC_ERRNO(ENOTRECOVERABLE),
+                    "Failed to load PIN in TPM: %s",
+                    sym_Tss2_RC_Decode(rc));
+            }
```

Comment:
coding style, no {} around single-line if blocks. also we indent continuation lines to double the usual indentation of 8ch